File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/ExampleScenario.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class ExampleScenario : DomainResource, IResource
{
    public string? Date { get; set; }
    public string? Publisher { get; set; }
    public ExampleScenario_Instance[]? Instance { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public string? Purpose { get; set; }
    public string? Name { get; set; }
    public ExampleScenario_Process[]? Process { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public bool? Experimental { get; set; }
    public string[]? Workflow { get; set; }
    public required string Status { get; set; }
    public string? Url { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? Version { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }
    public ExampleScenario_Actor[]? Actor { get; set; }

    public class ExampleScenario_Instance_Version : BackboneElement
    {
        public required string VersionId { get; set; }
        public required string Description { get; set; }
    }

    public class ExampleScenario_Instance_ContainedInstance : BackboneElement
    {
        public string? VersionId { get; set; }
        public required string ResourceId { get; set; }
    }

    public class ExampleScenario_Instance : BackboneElement
    {
        public string? Name { get; set; }
        public ExampleScenario_Instance_Version[]? Version { get; set; }
        public required string ResourceId { get; set; }
        public string? Description { get; set; }
        public required string ResourceType { get; set; }
        public ExampleScenario_Instance_ContainedInstance[]? ContainedInstance { get; set; }
    }

    public class ExampleScenario_Process_Step_Process_Step_Operation : BackboneElement
    {
        public string? Response { get; set; }
        public string? Description { get; set; }
        public string? Request { get; set; }
        public required string Number { get; set; }
        public string? Name { get; set; }
        public string? Initiator { get; set; }
        public string? Type { get; set; }
        public bool? ReceiverActive { get; set; }
        public bool? InitiatorActive { get; set; }
        public string? Receiver { get; set; }
    }

    public class ExampleScenario_Process_Step_Process_Step_Alternative : BackboneElement
    {
        public string[]? Step { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
    }

    public class ExampleScenario_Process_Step_Process_Step : BackboneElement
    {
        public bool? Pause { get; set; }
        public string[]? Process { get; set; }
        public ExampleScenario_Process_Step_Process_Step_Operation? Operation { get; set; }
        public ExampleScenario_Process_Step_Process_Step_Alternative[]? Alternative { get; set; }
    }

    public class ExampleScenario_Process_Step_Process : BackboneElement
    {
        public ExampleScenario_Process_Step_Process_Step[]? Step { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public string? PreConditions { get; set; }
        public string? PostConditions { get; set; }
    }

    public class ExampleScenario_Process_Step_Operation_Response : BackboneElement
    {
        public string? VersionId { get; set; }
        public required string ResourceId { get; set; }
    }

    public class ExampleScenario_Process_Step_Operation_Request : BackboneElement
    {
        public string? VersionId { get; set; }
        public required string ResourceId { get; set; }
    }

    public class ExampleScenario_Process_Step_Operation : BackboneElement
    {
        public ExampleScenario_Process_Step_Operation_Response[]? Response { get; set; }
        public string? Description { get; set; }
        public ExampleScenario_Process_Step_Operation_Request[]? Request { get; set; }
        public required string Number { get; set; }
        public string? Name { get; set; }
        public string? Initiator { get; set; }
        public string? Type { get; set; }
        public bool? ReceiverActive { get; set; }
        public bool? InitiatorActive { get; set; }
        public string? Receiver { get; set; }
    }

    public class ExampleScenario_Process_Step_Alternative_Step_Operation : BackboneElement
    {
        public string? Response { get; set; }
        public string? Description { get; set; }
        public string? Request { get; set; }
        public required string Number { get; set; }
        public string? Name { get; set; }
        public string? Initiator { get; set; }
        public string? Type { get; set; }
        public bool? ReceiverActive { get; set; }
        public bool? InitiatorActive { get; set; }
        public string? Receiver { get; set; }
    }

    public class ExampleScenario_Process_Step_Alternative_Step_Alternative : BackboneElement
    {
        public string[]? Step { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
    }

    public class ExampleScenario_Process_Step_Alternative_Step : BackboneElement
    {
        public bool? Pause { get; set; }
        public string[]? Process { get; set; }
        public ExampleScenario_Process_Step_Alternative_Step_Operation? Operation { get; set; }
        public ExampleScenario_Process_Step_Alternative_Step_Alternative[]? Alternative { get; set; }
    }

    public class ExampleScenario_Process_Step_Alternative : BackboneElement
    {
        public ExampleScenario_Process_Step_Alternative_Step[]? Step { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
    }

    public class ExampleScenario_Process_Step : BackboneElement
    {
        public bool? Pause { get; set; }
        public ExampleScenario_Process_Step_Process[]? Process { get; set; }
        public ExampleScenario_Process_Step_Operation? Operation { get; set; }
        public ExampleScenario_Process_Step_Alternative[]? Alternative { get; set; }
    }

    public class ExampleScenario_Process : BackboneElement
    {
        public ExampleScenario_Process_Step[]? Step { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public string? PreConditions { get; set; }
        public string? PostConditions { get; set; }
    }

    public class ExampleScenario_Actor : BackboneElement
    {
        public string? Name { get; set; }
        public required string Type { get; set; }
        public required string ActorId { get; set; }
        public string? Description { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/ExampleScenario.py
from typing import Optional, List
from base import UsageContext
from base import ContactDetail
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class ExampleScenario_Instance_Version(BackboneElement):
    version_id: str
    description: str

class ExampleScenario_Instance_ContainedInstance(BackboneElement):
    version_id: Optional[str] = None
    resource_id: str

class ExampleScenario_Instance(BackboneElement):
    name: Optional[str] = None
    version: Optional[List[ExampleScenario_Instance_Version]] = None
    resource_id: str
    description: Optional[str] = None
    resource_type: str
    contained_instance: Optional[List[ExampleScenario_Instance_ContainedInstance]] = None

class ExampleScenario_Process_Step_Operation(BackboneElement):
    response: Optional[Reference] = None
    description: Optional[str] = None
    request: Optional[Reference] = None
    number: str
    name: Optional[str] = None
    initiator: Optional[str] = None
    type: Optional[str] = None
    receiver_active: Optional[bool] = None
    initiator_active: Optional[bool] = None
    receiver: Optional[str] = None

class ExampleScenario_Process_Step_Alternative(BackboneElement):
    step: Optional[List[Reference]] = None
    title: str
    description: Optional[str] = None

class ExampleScenario_Process_Step(BackboneElement):
    pause: Optional[bool] = None
    process: Optional[List[Reference]] = None
    operation: Optional[ExampleScenario_Process_Step_Operation] = None
    alternative: Optional[List[ExampleScenario_Process_Step_Alternative]] = None

class ExampleScenario_Process(BackboneElement):
    step: Optional[List[ExampleScenario_Process_Step]] = None
    title: str
    description: Optional[str] = None
    pre_conditions: Optional[str] = None
    post_conditions: Optional[str] = None

class ExampleScenario_Actor(BackboneElement):
    name: Optional[str] = None
    type: str
    actor_id: str
    description: Optional[str] = None

class ExampleScenario(DomainResource):
    date: Optional[str] = None
    publisher: Optional[str] = None
    instance: Optional[List[ExampleScenario_Instance]] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    purpose: Optional[str] = None
    name: Optional[str] = None
    process: Optional[List[ExampleScenario_Process]] = None
    use_context: Optional[List[UsageContext]] = None
    copyright: Optional[str] = None
    experimental: Optional[bool] = None
    workflow: Optional[List[str]] = None
    status: str
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = None
    actor: Optional[List[ExampleScenario_Actor]] = None